src/goto-programs/write_goto_binary.cpp

```diff
@@ -97,11 +97,9 @@ bool write_goto_binary_v4(
         const goto_programt::instructiont &instruction = *i_it;
 
         irepconverter.reference_convert(instruction.code, out);
-        irepconverter.write_string_ref(out, instruction.function);
```

Comment:
I don't think we are there just yet: this should only be done once we have sanity checks of goto programs in place, and those ensure that `fct.first == instruction.function`, which I doubt truly holds at the moment.

Reply:
Ok, #3126!